In [1]:
import numpy as np
from matplotlib import pyplot as plt
from scipy.optimize import curve_fit
import scipy.optimize as spo

In [2]:
%matplotlib notebook

In [3]:
t = 2*10**-6

In [4]:
# Sheet resistance measurements
V1 = np.array([11.79, 11.81, 11.79, 11.80,
               32.83, 32.83, 32.81, 32.80]) * 10**-3
# Hall resistance measurements
V2 = np.array([21.21, 21.14, 21.10, 21.12]) * 10**-3
I = 3 * 10**-3

# Sheet resistance
R1 = V1/I
# Hall resistance
R2 = V2/I

print R1
print R2

[  3.93         3.93666667   3.93         3.93333333  10.94333333
  10.94333333  10.93666667  10.93333333]
[ 7.07        7.04666667  7.03333333  7.04      ]


In [5]:
x = 20

# Lab data
----------------------------------------------------------------------

## InSb $(t=2\mu m):$

#### Measuring the sheet resistance, $R_{sh}$

| Current I, (mA) |Voltage, $V_{ij,kl}$| Value, (mV) | Resistance ($\Omega$)
| --------------- |:------------:|:------------:|:------------:|
|       3         | $V_{14,23}$  | 11.79 | 3.93
|       3         | $V_{41,32}$  | 11.81 | 3.94
|       3         | $V_{23,14}$  | 11.79 | 3.93
|       3         | $V_{32,41}$  | 11.80 | 3.93
|       3         | $V_{12,43}$  | 32.83 | 10.94
|       3         | $V_{21,34}$  | 32.83 | 10.94
|       3         | $V_{43,12}$  | 32.81 | 10.94
|       3         | $V_{34,21}$  | 32.80 | 10.93

#### Measuring the Hall Resistance
| Current I, (mA) |Voltage, $V_{ij,kl}$| Value, (mV) | Resistance ($\Omega$)
| --------------- |:------------:|:------------:|:------------:|
|       3         | $V_{13,24}$  | 21.21 | 7.07
|       3         | $V_{31,42}$  | 21.14 | 7.05
|       3         | $V_{24,13}$  | 21.10 | 7.03
|       3         | $V_{42,31}$  | 21.12 | 7.04

In [8]:
Ra = np.average(R1[4:-1])
Rb = np.average(R1[0:4])
ratio = 32.81/11.79

In [9]:
def f(y, x=1.408):
    return np.cosh((x-1)/(x+1) * np.log(2)/y) - 0.5*np.exp(np.log(2)/y)

In [10]:
x0 = ratio
z = spo.brentq(f, 0.01, 0.99, (x0))

In [11]:
R_sh = (np.pi / np.log(2)) * ((Ra+Rb)/2) * z
rho = R_sh * t
print 'rho = ' + str(rho)+' Ohm m'
print 'sigma = ' + str(rho**-1)+' W / K*m'

rho = 6.18739360222e-05 Ohm m
sigma = 16161.8940751 W / K*m


In [12]:
1 / (rho*100)

161.61894075091254

$\rho = 6.19x10^{-3} \Omega {cm}^{-1}$

$\sigma = 161.6 W K^{-1} {cm}^{-1}$

In [13]:
print R_sh

30.9369680111


#### $R_{H}$ : Measuring the Hall coefficient, for fixed magnetic field strength

Positive Field, Positive Current

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 100.2   | 57.490
|       2.70         | 100.2   | 51.755
|       2.40         | 100.2   | 46.035
|       2.10         | 100.2   | 40.310
|       1.80         | 100.2   | 34.573
|       1.50         | 100.3   | 28.865
|       1.20         | 100.3   | 23.187
|       0.90         | 100.3   | 17.476
|       0.60         | 100.3   | 11.659
|       0.30         | 100.5   | 5.840


Positive Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 100.2   | -56.734
|       2.70         | 100.2   | -51.064
|       2.40         | 99.7   | -45.430
|       2.10         | 99.7   | -39.775
|       1.80         | 99.8   | -34.112
|       1.50         | 99.7   | -28.430
|       1.20         | 99.8   | -22.764
|       0.90         | 99.8   | -17.073
|       0.60         | 99.8   | -11.389
|       0.30         | 100.1   | -5.715

Negative Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -100.3   | 57.540
|       2.70         | -100.3   | 51.813
|       2.40         | -100.3   | 46.077
|       2.10         | -100.3   | 40.330
|       1.80         | -99.4   | 34.582
|       1.50         | -99.4   | 28.827
|       1.20         | -99.4   | 23.065
|       0.90         | -99.4   | 17.303
|       0.60         | -99.4   | 11.544
|       0.30         | -99.9   | 5.813

Negative Field, Positive current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -100.3   | -59.040
|       2.70         | -100.3   | -53.191
|       2.40         | -101.4   | -47.309
|       2.10         | -101.4   | -41.427
|       1.80         | -101.4   | -35.540
|       1.50         | -101.4   | -29.640
|       1.20         | -101.3   | -23.738
|       0.90         | -101.3   | -17.818
|       0.60         | -101.2   | -11.905
|       0.30         | -100.2   | -5.983


-------------------------------------------


Positive Field, Positive Current

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 149.1   | 84.365
|       2.70         | 149.1   | 75.968
|       2.40         | 149.2   | 67.555
|       2.10         | 149.2   | 59.161
|       1.80         | 149.2   | 50.737
|       1.50         | 149.2   | 42.303
|       1.20         | 149.2   | 33.870
|       0.90         | 149.2   | 25.412
|       0.60         | 149.2   | 16.962
|       0.30         | 149.2   | 8.498

Positive Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 150.1   | -84.170
|       2.70         | 150.1   | -75.767
|       2.40         | 150.1   | -67.374
|       2.10         | 150.1   | -58.986
|       1.80         | 150.1   | -50.581
|       1.50         | 150.1   | -42.163
|       1.20         | 150.1   | -33.744
|       0.90         | 150.1   | -25.311
|       0.60         | 150.0   | -16.886
|       0.30         | 149.1   | -8.453

Negative Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -150.9   | 87.620
|       2.70         | -150.9   | 78.875
|       2.40         | -150.9   | 70.141
|       2.10         | -150.9   | 61.403
|       1.80         | -150.9   | 52.663
|       1.50         | -150.8   | 43.899
|       1.20         | -150.8   | 35.141
|       0.90         | -150.8   | 26.370
|       0.60         | -150.8   | 17.600
|       0.30         | -150.7   | 8.818

Negative Field, Positive current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -150.5   | -87.31
|       2.70         | -150.8   | -78.610
|       2.40         | -150.8   | -69.904
|       2.10         | -150.8   | -63.894
|       1.80         | -150.8   | -52.473
|       1.50         | -150.8   | -43.738
|       1.20         | -150.8   | -35.012
|       0.90         | -150.8   | -26.262
|       0.60         | -150.8   | -17.520
|       0.30         | -150.9   | -8.770


-------------------------------------------

Positive Field, Positive Current

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 198.5   | 111.413
|       2.70         | 198.5   | 100.294
|       2.40         | 198.5   | 89.181
|       2.10         | 198.5   | 78.080
|       1.80         | 198.5   | 66.954
|       1.50         | 198.5   | 55.820
|       1.20         | 198.5   | 44.688
|       0.90         | 198.5   | 33.530
|       0.60         | 198.5   | 22.384
|       0.30         | 198.5   | 11.224

Positive Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 198.5   | -111.111
|       2.70         | 198.5   | -100.039
|       2.40         | 198.5   | -88.948 
|       2.10         | 198.5   | -77.853
|       1.80         | 198.5   | -66.749
|       1.50         | 198.5   | -55.641
|       1.20         | 198.5   | -44.530
|       0.90         | 198.5   | -33.404
|       0.60         | 198.5   | -22.285
|       0.30         | 198.5   | -11.155

Negative Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -198.9   | 113.50
|       2.70         | -198.8   | 102.16
|       2.40         | -198.8   | 90.90
|       2.10         | -198.9   | 79.576
|       1.80         | -198.9   | 68.246
|       1.50         | -198.9   | 56.915 
|       1.20         | -198.8   | 45.572
|       0.90         | -199.9   | 34.208
|       0.60         | -199.9   | 22.863
|       0.30         | -198.9   | 12.094

Negative Field, Positive current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -198.8   | -113.17
|       2.70         | -198.8   | -101.87
|       2.40         | -198.9   | -90.584
|       2.10         | -198.9   | -79.286
|       1.80         | -198.9   | -67.989
|       1.50         | -198.8   | -56.670
|       1.20         | -198.9   | -45.359
|       0.90         | -198.9   | -34.028
|       0.60         | -198.8   | -22.697
|       0.30         | -198.8   | -11.361


-------------------------------------------

Positive Field, Positive Current

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 249.7   | 136.24
|       2.70         | 249.7   | 122.662
|       2.40         | 249.7   | 109.071
|       2.10         | 249.7   | 93.530
|       1.80         | 249.7   | 81.874
|       1.50         | 249.7   | 68.257
|       1.20         | 249.7   | 54.630
|       0.90         | 249.7   | 40.988
|       0.60         | 249.7   | 27.351
|       0.30         | 249.7   | 13.707

Positive Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 249.9   | -135.893
|       2.70         | 249.8   | -122.347
|       2.40         | 249.7   | -108.781
|       2.10         | 249.7   | -95.220
|       1.80         | 249.5   | -81.672
|       1.50         | 249.8   | -68.094
|       1.20         | 249.3   | -54.496
|       0.90         | 249.2   | -40.879
|       0.60         | 249.1   | -27.267
|       0.30         | 249.7   | -13.648

Negative Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -249.9   | 139.70
|       2.70         | -249.9   | 125.35
|       2.40         | -249.7   | 111.46
|       2.10         | -249.6   | 97.57
|       1.80         | -249.5   | 83.670
|       1.50         | -250.2   | 69.775
|       1.20         | -250.2   | 55.845
|       0.90         | -250.1   | 41.910
|       0.60         | -250.1   | 28.004
|       0.30         | -250.1   | 14.710

Negative Field, Positive current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -249.9   | -137.48
|       2.70         | -249.9   | -123.76
|       2.40         | -249.9   | -110.06
|       2.10         | -249.9   | -96.34
|       1.80         | -249.9   | -82.611
|       1.50         | -249.9   | -68.866
|       1.20         | -249.9   | -55.120
|       0.90         | -249.9   | -41.355
|       0.60         | -249.9   | -27.585
|       0.30         | -249.9   | -13.811



-------------------------------------------

Positive Field, Positive Current

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 301   | 158.02
|       2.70         | 301   | 142.27
|       2.40         | 301   | 126.520
|       2.10         | 301 | 110.758
|       1.80         | 301   | 94.978
|       1.50         | 301   | 79.182
|       1.20         | 302     | 63.394
|       0.90         | 302   | 47.584
|       0.60         | 302   | 31.761
|       0.30         | 302   | 15.928

Positive Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | 302   | -157.64
|       2.70         | 302   | -141.94
|       2.40         | 302   | -126.227
|       2.10         | 303   | -110.487
|       1.80         | 301   | -94.745
|       1.50         | 302   | -78.975
|       1.20         | 303   | -63.199
|       0.90         | 301   | -47.412
|       0.60         | 302   | -31.630
|       0.30         | 302   | -15.833

Negative Field, Negative current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -301   | 161.28
|       2.70         | -301   | 145.20
|       2.40         | -301   | 129.13
|       2.10         | -301   | 113.05
|       1.80         | -301   | 96.946
|       1.50         | -301   | 80.823
|       1.20         | -301   | 64.695
|       0.90         | -301   | 48.543
|       0.60         | -301   | 32.389
|       0.30         | -301   | 16.221

Negative Field, Positive current:

| Current I, (mA) |Magnetic Field B, (mT) | Voltage, V_H (mV)
| --------------- |:------------:|:------------:|
|       3.00         | -301   | -160.71
|       2.70         | -301   | -144.70
|       2.40         | -301   | -128.68
|       2.10         | -301   | -112.64
|       1.80         | -301   | -96.586
|       1.50         | -301   | -80.521
|       1.20         | -301   | -64.439
|       0.90         | -301   | -48.342
|       0.60         | -301   | -32.252
|       0.30         | -301   | -16.146

In [14]:
I = np.arange(0.3,3.3,0.3)*10**-3
V1 = np.array([5.840, 11.659, 17.476, 23.187, 28.865, 34.573, 40.310, 46.035, 51.755, 57.490]) * 10**-3
V2 = np.array([-5.715, -11.389, -17.073, -22.764, -28.43, -34.112, -39.775, -45.430, -51.064, -56.734]) * 10**-3
V3 = np.array([5.813, 11.544, 17.303, 23.065, 28.827, 34.582, 40.330, 46.077, 51.813, 57.540]) * 10**-3
V4 = np.array([-5.983, -11.905, -17.818, -23.738, -29.640, -35.540, -41.427, -47.309, -53.191, -59.040]) * 10**-3
V = (V1 - V2 + V3 - V4)/4
B1 = np.array([100.5, 100.3, 100.3, 100.3, 100.3, 100.2, 100.2, 100.2, 100.2, 100.2]) * 10**-3
B2 = np.array([100.1, 99.8, 99.8, 99.8, 99.7, 99.8, 99.7, 99.7, 100.2, 100.2]) * 10**-3
B3 = np.array([99.9, 99.4, 99.4, 99.4, 99.4, 99.4, 100.3, 100.3, 100.3, 100.3]) * 10**-3
B4 = np.array([-101.2, -101.2, -101.3, -101.3, -101.4, -101.4, -101.4, -101.4, -101.3, -101.3]) * 10**-3
B = 100*10**-3
IB = I*B
VB1 = V1/B1
VB2 = V2/B2
VB3 = V3/B3
VB4 = V4/B4
VB = (VB1 - VB2 + VB3 - VB4)/4

In [15]:
fig1 = plt.figure()
ax = fig1.add_subplot(111)
ax.set_title('A plot of Current against Voltage, at fixed magnetic field strength')
#ax.legend('For fixed B=0.374mT')
ax.plot(V*1000, IB*1000)
ax.set_xlabel("Current x B-field strength IB / (A mT)")
ax.set_ylabel("Voltage / (mV)")
print V

FigureCanvasNbAgg()

[ 0.00583775  0.01162425  0.0174175   0.0231885   0.0289405   0.03470175
  0.0404605   0.04621275  0.05195575  0.057701  ]


In [16]:
def y(x, A, B):
    return A*x + B
R_h_eff,foo = curve_fit(y, IB, V)[0]

In [17]:
print 'R_h effective= ' + str(R_h_eff) + ' Ohm/Tesla'

R_h effective= 192.04914137 Ohm/Tesla


In [18]:
mu = R_h_eff/rho * t
print 'mu = ' + str(mu) + ' m^2/Vs'
n = (R_h_eff*t*1.67*10**-19)**-1
print 'n = ' + str(n) + ' m^-3'

mu = 6.20775576006 m^2/Vs
n = 1.55898222439e+22 m^-3


$\mu = 62078 {cm}^{2}V^{-1}s^{-1}$

$n = 1.56x10^{16} {cm}^{-3}$

In [19]:
B

0.1